game/songs.cc

```diff
@@ -381,7 +403,21 @@ void Songs::sortChange(int diff) {
 	if (m_order < 0) m_order += orders;
 	RestoreSel restore(*this);
 	config["songs/sort-order"].i() = m_order;
+	switch (m_order) {
+		case 1:
```

Comment:
add //fallthrough, or //falls through to indicate this was on purpose and to suppress -Wfallthrough from GCC

Reply:
Should I use ```[[fallthrough]]```? (will that work, being c++17?) or literally ```//fallthrough```

Reply:
a simple comment is good enough (at least for GCC).
from the manual:
https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html

C++17 provides a standard way to suppress the -Wimplicit-fallthrough warning using [[fallthrough]]; instead of the GNU attribute. In C++11 or C++14 users can use [[gnu::fallthrough]];, which is a GNU extension. Instead of these attributes, it is also possible to add a fallthrough comment to silence the warning. The whole body of the C or C++ style comment should match the given regular expressions listed below. The option argument n specifies what kind of comments are accepted:

    -Wimplicit-fallthrough=0 disables the warning altogether.
    -Wimplicit-fallthrough=1 matches .* regular expression, any comment is used as fallthrough comment.
    -Wimplicit-fallthrough=2 case insensitively matches .*falls?[ \t-]*thr(ough|u).* regular expression.
    -Wimplicit-fallthrough=3 case sensitively matches one of the following regular expressions:
        -fallthrough
        @fallthrough@
        lint -fallthrough[ \t]*
        [ \t.!]*(ELSE,? |INTENTIONAL(LY)? )?
        FALL(S | |-)?THR(OUGH|U)[ \t.!]*(-[^\n\r]*)?
        [ \t.!]*(Else,? |Intentional(ly)? )?
        Fall((s | |-)[Tt]|t)hr(ough|u)[ \t.!]*(-[^\n\r]*)?
        [ \t.!]*([Ee]lse,? |[Ii]ntentional(ly)? )?
        fall(s | |-)?thr(ough|u)[ \t.!]*(-[^\n\r]*)? 
    -Wimplicit-fallthrough=4 case sensitively matches one of the following regular expressions:
        -fallthrough
        @fallthrough@
        lint -fallthrough[ \t]*
        [ \t]*FALLTHR(OUGH|U)[ \t]* 
    -Wimplicit-fallthrough=5 doesn’t recognize any comments as fallthrough comments, only attributes disable the warning. 

The comment needs to be followed after optional whitespace and other comments by case or default keywords or by a user label that precedes some case or default label.

switch (cond)
  {
  case 1:
    bar (0);
    /* FALLTHRU */
  default:
    …
  }